File: web-portal/back-center-vue/src/views/index/system/menu/interface/sys-menu.ts
import { TreeData } from 'element-ui/types/tree'

export enum IsHiddenMenu {
  是 = 1,
  否 = 0
}
export enum IsMenu {
  是 = 1,
  否 = 2
}

export type Name = Model.Name
export interface RoleIdAndMenuIds {
  menuIds: Array<number> | null
  roleId: number | null
}
export interface MenuForm extends Name, Model.ParentId {
  css: string
  isHidden: IsHiddenMenu | boolean | string
  isMenu: IsMenu | number | string
  path: string
  sort: number
  url: string
  title: string
}
export interface MenuResponse extends Model.BaseField, Name, RoleIdAndMenuIds, MenuForm {
  subMenus: Array<MenuResponse> | null
}
export interface Id extends Name {
  id: number
}

export interface IMenuTree extends Name, TreeData {
  children: Array<IMenuTree>
}

export interface MenuRole extends Id {
  checked: boolean
  pId: number
  open: boolean
}

File: inner-intergration/common-spring-boot-starter/src/main/java/com/key/win/common/model/system/SysMenu.java
package com.key.win.common.model.system;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.impl.StringArraySerializer;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import com.key.win.common.model.basic.MybatisID;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

/**
* 类说明  菜单实体
*/
@ApiModel("菜单实体")
@Data
@TableName("sys_menu")
@EqualsAndHashCode(callSuper=true)
public class SysMenu  extends MybatisID {

	@ApiModelProperty("父节点Id")
	@JsonSerialize(using = ToStringSerializer.class)
	private Long parentId;
	@ApiModelProperty("名称")
	private String name;
	@ApiModelProperty("标题")
	private String title;
	@ApiModelProperty("url")
	private String url;
	@ApiModelProperty("路径")
	private String path;
	@ApiModelProperty("css")
	private String css;
	@ApiModelProperty("排序")
	private Integer sort;
	@ApiModelProperty("是否菜单：1-目录，2-菜单")
	private Integer isMenu;
	@ApiModelProperty("是否隐藏：true-是 false-否")
	private Boolean isHidden;
	@ApiModelProperty("孩子列表")
	@TableField(exist = false)
	private List<SysMenu> subMenus = new ArrayList<>();
	@ApiModelProperty("菜单Id集合")
	@TableField(exist = false)
	@JsonSerialize(using = StringArraySerializer.class)
	private Set<Long> menuIds;

	public void addSubMenu(SysMenu sysMenu) {
		this.subMenus.add(sysMenu);
	}

}
